### ISSUE TITLE
Screen sharing windows doesn't show up

### ISSUE THREAD

**Issue by SprainedSpark89** [smartfrigde/armcord#565]
**Describe the bug**
When pressing the screen share button nothing happens

**To Reproduce**
1. Join a VC
2. Click Share Screen

**Expected behavior**
To show the screen share window like it did when i was using pi os 32 bit legacy

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Pi OS 64 bit
 - Method of installation Pi Apps
 - Version 3.2.6

**Additional context**
Everything I've been doing in this new stupid os seems to be broken or not work...

**Comment by Heus-Sueh** [comment]
the same thing happens to me on sway wm

**Comment by SomeAspy** [comment]
Does this look more similar to your issue? https://github.com/ArmCord/ArmCord/issues/594

**Comment by Heus-Sueh** [comment]
> Does this look more similar to your issue? #594

Mine doesn't even show the sharing screen, using Vesktop it actually works

**Comment by SprainedSpark89** [comment]
> Does this look more similar to your issue? #594
No, there's not even a prompt when clicking the share screen button.

**Comment by smartfrigde** [comment]
Try enabling legacy capturer in settings

**Comment by SprainedSpark89** [comment]
> Try enabling legacy capturer in settings

Nope, doesn't work.

**Comment by smartfrigde** [comment]
Are you using Wayland?

**Comment by SprainedSpark89** [comment]
no, unless pi os 64 bit changed that

**Comment by SomeAspy** [comment]
> no, unless pi os 64 bit changed that

I think it does
https://www.omglinux.com/raspberry-pi-os-bookworm/

**Comment by SprainedSpark89** [comment]
well i am using wayland, didn't know that changed with pi os 64 bit

**Comment by Heus-Sueh** [comment]
> > Does this look more similar to your issue? #594
> 
> Mine doesn't even show the sharing screen, using Vesktop it actually works

I reported this problem a while ago, but I've been able to share my screen for some time now. However, the window selection screen still doesn't appear. I believe this is a limitation of [xdg-desktop-portal-wlr](https://github.com/emersion/xdg-desktop-portal-wlr). Any desktop environment or window manager that isn't based on GNOME or KDE suffers from this limitation. Hyprland is an exception since they implemented their own version of xdg-desktop-portal with some hacks. Only recently, Wayland received an official protocol for capturing individual windows: https://www.phoronix.com/news/Wayland-Merges-Screen-Capture.

**Comment by Heus-Sueh** [comment]
> > no, unless pi os 64 bit changed that
> 
> I think it does [omglinux.com/raspberry-pi-os-bookworm](https://www.omglinux.com/raspberry-pi-os-bookworm/)

Searching here, I saw that the pi os desktop is based on the wayfire compositor (made with wlroots) sway is also made with wlroots, I think pi os also uses the same xdg-desktop-portal that I mentioned above

**Comment by hashcube-dev** [comment]
>  Hyprland is an exception since they implemented their own version of xdg-desktop-portal with some hacks.

Hyprland's xdg stuff used to work great, but now I'm experiencing this exact bug (I think???). Here is some Ran From The Terminal Info
```
{
  frame: WebFrameMain {
    _events: [Object: null prototype] {},
    _eventsCount: 0
  },
  securityOrigin: 'https://discord.com/',
  userGesture: true,
  videoRequested: true,
  audioRequested: true
}
[378572:0530/171457.997430:ERROR:screen_capture_portal_interface.cc(48)] Failed to request session: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.ScreenCast” on object at path /org/freedesktop/portal/desktop
[378572:0530/171457.997442:ERROR:base_capturer_pipewire.cc(81)] ScreenCastPortal failed: 3
Failed to get sources.
(node:378572) UnhandledPromiseRejectionWarning: TypeError: Video was requested, but no video stream was provided
    at file:///nix/store/hmppzmj0l0vp08455nh0w0jn9szxxd4x-legcord-1.1.4/share/lib/legcord/resources/app.asar/ts-out/main-GajYH2d9.js:43:545
(node:378572) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 18)
```

So it definitely is xdg, but I have no clue why it stopped working on my hyprland.
